lib/matplotlib/tests/test_widgets.py

```diff
@@ -254,6 +254,27 @@ def test_CheckButtons():
     check.disconnect(cid)
 
 
+def check_TextBox():
+    def submit(text):
+        tool.set_val('x**1')
+
+    def change(text):
+        tool.color = '1.0'
+    ax = get_ax()
+    tool = widgets.TextBox(ax, 'Evaluate', color='.95', initial='x**2')
+    assert tool.text == 'x**2'
+    tool.on_submit(submit)
+    tool.on_text_change(change)
+    tool.begin_typing(tool.text)
+    tool.stop_typing()
+    assert tool.text == 'x**1'
+    assert tool.color == '1.0'
+
+
+def test_TextBox():
```

Comment:
Is there a reason to nest these like this rather than just call the above `test_TextBox`?

Reply:
Not particularly, I have changed that now. Thanks!